cartridges/int_algolia/cartridge/scripts/algolia/steps/sendChunkOrientedProductUpdates.js

```diff
@@ -53,37 +52,44 @@ exports.beforeStep = function(parameters, stepExecution) {
     logger = require('dw/system/Logger').getLogger('algolia', 'Algolia');
     productFilter = require('*/cartridge/scripts/algolia/filters/productFilter');
     algoliaProductConfig = require('*/cartridge/scripts/algolia/lib/algoliaProductConfig');
+    AlgoliaJobReport = require('*/cartridge/scripts/algolia/helper/AlgoliaJobReport');
 
-    // checking mandatory parameters
-    if (empty(parameters.resourceType)) {
-        let errorMessage = 'Mandatory job step parameter "resourceType" missing!';
-        jobHelper.logError(errorMessage);
-        return;
+    /* --- initializing custom object logging --- */
+    jobReport = new AlgoliaJobReport(stepExecution.getJobExecution().getJobID(), 'product');
+    jobReport.startTime = new Date();
+
+
+    /* --- parameters --- */
+    paramFieldListOverride = algoliaData.csvStringToArray(parameters.fieldListOverride); // fieldListOverride - pass it along to sending method
+    paramIndexingMethod = parameters.indexingMethod || 'partialRecordUpdate'; // 'partialRecordUpdate' (default), 'fullRecordUpdate' or 'fullCatalogReindex'
+
+
+    /* --- fieldListOverride parameter --- */
+    if (empty(paramFieldListOverride)) {
+        const customFields = algoliaData.getSetOfArray('CustomFields');
+        fieldsToSend = algoliaProductConfig.defaultAttributes.concat(customFields);
+    } else {
+        fieldsToSend = paramFieldListOverride;
     }
+    logger.info('fieldListOverride parameter: ' + paramFieldListOverride);
+    logger.info('Actual fields to be sent: ' + JSON.stringify(fieldsToSend));
 
-    // parameters
-    resourceType = parameters.resourceType; // resouceType ( price | inventory | product ) - pass it along to sendChunk()
-    fieldListOverride = algoliaData.csvStringToArray(parameters.fieldListOverride); // fieldListOverride - pass it along to sendChunk()
-    indexingMethod = parameters.indexingMethod || 'partialRecordUpdate';
 
-    switch (indexingMethod) {
+    /* --- indexingMethod parameter --- */
+    switch (paramIndexingMethod) {
         case 'fullRecordUpdate':
-        case 'fullCatalogReindexUpdate':
```

Comment:
I believe this was a mistake when you renamed the options for the `indexingMethod` parameter, let me know otherwise